test/cpp/api/autograd.cpp

```diff
@@ -154,6 +155,40 @@ TEST(AutogradAPITests, RetainGrad) {
   ASSERT_VARIABLE_EQ(input * 18, input.grad());
 }
 
+TEST(AutogradAPITests, AnomalyMode) {
+  // Needs to have backtrace as warning and then throw an error
+  torch::autograd::AnomalyMode::set_enabled(true);
+  {
+    WarningCapture warnings;
+    auto x = torch::tensor({5.0}, torch::requires_grad());
+    auto y = x * x;
+    auto z = y * y;
+    y += 1;
+    ASSERT_THROWS_WITH(z.backward(), "inplace");
+    ASSERT_TRUE(
+        warnings.str().find("Traceback of forward") != std::string::npos);
+  }
+  {
+    WarningCapture warnings;
+    // Double backward
+    auto x = torch::tensor({0.0}, torch::requires_grad());
+    auto y = x.pow(1.5);
+    auto gr =
+        grad({y}, {x}, {}, /*retain_graph=*/true, /*create_backward=*/true);
+    ASSERT_THROWS_WITH(grad({gr[0]}, {x});, "returned nan");
```

Comment:
@t-vi this should return inf and not nan. It returns nan on master due to this change here https://github.com/pytorch/pytorch/pull/44444/files#diff-4adbd88239afcd60e8198aab65d4f5e43b62314e34b80551e997a1ea503adea5R188 where the exp type is set to `c10::complex<double>` always. I am gonna comment this test in https://github.com/pytorch/pytorch/pull/46639 to fix this behavior first. can you re-add this test back with a case where nan would be returned ...?

Reply:
So if the test would use a complex 0, the test would continue to be nan?
If you had the chance I'd appreciate if you updated the test instead of disabling. Thank you.

Reply:
right makes sense! will do :D